modules/host/negotiate_test.go

```diff
@@ -0,0 +1,88 @@
+package host
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/NebulousLabs/Sia/consensus"
+	"github.com/NebulousLabs/Sia/modules"
+)
+
+// testAllocation allocates and then deallocates a file, checking that the
+// space is returned and the file is actually deleted.
+func (ht *HostTester) testAllocation() {
+	currentSpaceRemaining := ht.spaceRemaining
+
+	// Allocate a 4kb file.
+	file, path, err := ht.allocate(4e3)
+	if err != nil {
+		ht.Fatal(err)
+	}
+	file.Close()
+
+	// Check that the file has a real name and that it exists on disk.
+	if path == "" {
+		ht.Fatal("path of allocated file is empty!")
+	}
+	fullpath := filepath.Join(ht.Host.hostDir, path)
+	_, err = os.Stat(fullpath)
+	if os.IsNotExist(err) {
+		ht.Fatal("file does not exist on disk")
+	}
+
+	// Check that spaceRemaining has decreased appropriately.
+	if currentSpaceRemaining != ht.spaceRemaining+4e3 {
+		ht.Error("space remaining did not decrease appropriately after allocating a file")
+	}
```

Comment:
this condition confuses me. How about:
```go
const filesize = 4e3 // use an untyped const, otherwise Go will think you want a float

initialSpace := ht.spaceRemaining
//...
if ht.spaceReamaining != initialSpace - filesize {
```